File: 3335. Total Characters in String After Transformations I/lengthAfterTransformations.js
/**
 * @param {string} s
 * @param {number} t
 * @return {number}
 */

 //ASCII for z => 122
 //ASCII for a => 97
 //Alphabet character count is => 26
 var lengthAfterTransformations = function(s, t) {
    const MOD = 1000000007
    let chars = new Array(26).fill(0)

    //count each character appearance
    for(let char of s){
        chars[char.charCodeAt(0)-97]++
    }

    //loop "t" times
    for(let i=0; i<t ;i++){
        let visit = new Array(26).fill(0)

        for(let j=0 ; j<26; j++){
            //when j reach "z"at index 25
            //increment at index 0 and 1 for "a"  and "b" by the cnt of "z"
            if(j === 25){
                visit[0] = (visit[0] + chars[j]) % MOD
                visit[1] = (visit[1] + chars[j]) % MOD
            }
            else{
                //transfered character
                //the init count in "chars" plus the previous character
                visit[j+1] =  (visit[j+1] + chars[j]) % MOD
            }
        }
        chars = visit
    }
    return chars.reduce((acc, curr)=> acc+curr,0)% MOD
};

File: 3335. Total Characters in String After Transformations I/lengthAfterTransformations.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
    public:
        int lengthAfterTransformations(string s, int t) {
            //count each character appearance
            vector<long long> chars(26,0);
            long long MOD = 1000000007;
            long long ans = 0;
    
            for(auto item : s){
                chars[item-97]++;
            }
    
            //loop "t" times
            for(int i=0; i<t; i++){
                //count the update for every alphabit change will loop 't' time
                vector<long long> visit(26,0);
    
                //loop through 26 charactes
                for(int ch=0; ch<26; ch++){
                    //if we reach "z" so... 
                    // => increment "a" count
                    // => increment "b" count 
                    //by "z" origin  (updated) count
                    if(ch == 25){
                        //for "a"
                        visit[0] = (visit[0] + chars[25]) % MOD;
                        //for "b"
                        visit[1] = (visit[1] + chars[25]) % MOD;
                    }
                    else{
                        //increment the transformed (new) character count by the previous charcate count in (chars) array
                        visit[ch+1] = (visit[ch+1] + chars[ch]) % MOD;
                    }
                }
    
                //update the chars vector data
                chars = visit;
            }
    
    
            //sum all together
            for(auto num : chars){
                ans += num;
            }
            return ans % MOD;
        }
    };